lms/djangoapps/course_blocks/transformers/hidden_content.py

```diff
@@ -111,5 +89,9 @@ def _is_block_hidden(self, block_structure, block_key):
         if self_paced:
             hidden_date = block_structure[block_structure.root_block_usage_key].end
         else:
-            hidden_date = self._get_merged_due_date(block_structure, block_key)
+            # Important Note:
+            # A small subtlety of grabbing the due date here is that this transformer relies on the
+            # DateOverrideTransformer (located in edx-when repo) to first set any overrides (one
+            # example is a user receiving an extension on an assignment).
+            hidden_date = block_structure.get_xblock_field(block_key, 'due', None) or MAXIMUM_DATE
```

Comment:
MAXIMUM_DATE in this case will just mean it's allowed. Maintaining the same behavior as before with the merged_due_date.